single_kernel_mongo/events/lifecycle.py

```diff
@@ -161,7 +164,12 @@ def on_config_changed(self, event: ConfigChangedEvent):
         """Config Changed Event."""
         try:
             self.dependent.update_config_and_restart()
-        except (UpgradeInProgressError, WaitingForLeaderError):
+        except (
+            UpgradeInProgressError,
+            WaitingForLeaderError,
+            DeferrableFailedHookChecksError,
+            SetPasswordError,
+        ):
             event.defer()
```

Comment:
Realising now, we should log a line here to explain that we're deferring and for what reason (something like [that](https://github.com/canonical/mongo-single-kernel-library/blob/6/edge/single_kernel_mongo/events/sharding.py#L154).

You can use the helper method `defer_event_with_info_log` that you can find in `single_kernel_mongo.utils.event_helpers` that will log a formatted line and defer.